src/main/java/mate/academy/rickandmorty/controller/CharacterController.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.rickandmorty.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.internal.CharacterDto;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/characters")
+@RequiredArgsConstructor
+@Tag(name = "Character management",
+        description = "Endpoint for getting Rick and Morty characters")
+public class CharacterController {
+    private final CharacterService characterService;
+
+    @GetMapping("/search")
```

Comment:
```suggestion
    @GetMapping
```
that's more RESTful naming - we should focus on hierarchy of resources and avoid verbs. HTTP verbs should express the idea - GET is retrieving, searching